daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/layers/LayerTransformerFactory.scala

```diff
@@ -17,15 +17,49 @@
 
 package org.apache.daffodil.runtime1.layers
 
+import org.apache.daffodil.runtime1.layers.api.Layer
+import org.apache.daffodil.runtime1.layers.api.LayerCompileInfo
+import org.apache.daffodil.runtime1.layers.api.LayerException
+
 /**
  * Factory for a layer transformer.
  *
  * This is the serialized object which is saved as part of a processor.
  * It constructs the layer transformer at runtime when newInstance() is called.
  *
- * This must be implmented as part of implementation of a daffodil layer.
+ * This allows the layer instance itself to be stateful.
  */
-abstract class LayerTransformerFactory(val name: String) extends Serializable {
+class LayerTransformerFactory(val layerCompileInfo: LayerCompileInfo) extends Serializable {
```

Comment:
Wondered if we could get rid of this in another comment. If we don't, I'd suggest we rename these transformer classes. I'm not sure they really do any transforming anymore, they really just a wrappers to allow Layers to store state in the `Layer` instance. Maybe `LayerWrapper` or something?